hera_sim/adjustment.py

```diff
@@ -506,43 +504,30 @@ def interpolate_to_reference(
 
     target_is_simulator = isinstance(target, Simulator)
     target = _to_uvdata(target)
-    target_time_to_lst_map = {
-        target_time: target_lst
-        for target_time, target_lst in zip(target.time_array, target.lst_array)
-    }
-    target_lsts = np.array(list(target_time_to_lst_map.values()))
+    target_times, inds = np.unique(target.time_array, return_index=True)
+    target_lsts = target.lst_array[inds].copy()
     target_freqs = np.unique(target.freq_array)
 
-    # TODO: figure out how to handle phase wraps
     def iswrapped(lsts):
         return np.any(lsts < lsts[0])
 
-    if axis in ("time", "both"):
-        if iswrapped(target_lsts) or iswrapped(ref_lsts):
-            raise NotImplementedError(
-                "Either the target LSTs or the reference LSTs have a phase wrap. "
-                "This is currently not supported."
-            )
-
     # Ensure reference parameters are a subset of target parameters.
     if axis in ("time", "both"):
-        if np.any(np.logical_or(ref_lsts < target_lsts[0], ref_lsts > target_lsts[-1])):
+        # Unwrap the LST axis if we have a phase wrap.
+        if iswrapped(target_lsts):
+            target_lsts[target_lsts < target_lsts[0]] += 2 * np.pi
+        if iswrapped(ref_lsts):
+            ref_lsts[ref_lsts < ref_lsts[0]] += 2 * np.pi
+
+        if np.any((ref_lsts < target_lsts[0]) | (ref_lsts > target_lsts[-1])):
             warn("Reference LSTs not a subset of target LSTs; clipping.")
-            key = np.argwhere(
-                np.logical_and(target_lsts[0] <= ref_lsts, ref_lsts <= target_lsts[-1])
-            ).flatten()
+            key = (target_lsts[0] <= ref_lsts) & (ref_lsts <= target_lsts[-1])
```

Comment:
I don't love calling this `key` since it's not a dictionary key but more like a boolean slicing operation. I'd call it something more descriptive like `is_in_range`